.github/workflows/build.yml

```diff
@@ -465,7 +465,7 @@ jobs:
         - name: GCC Version
           run: gcc --version # If this fails MINGW is not setup correctly
         - name: Configure
-          run: cmake -S. -B build -D UPDATE_DEPS=ON -D CMAKE_BUILD_TYPE=Release -D ENABLE_WERROR=ON
```

Comment:
Add a comment explaining why USE_MASM=OFF for MinGW builds.

Reply:
This is for the `mingw-no-asm` build. Should I re-enable and test the normal `mingw` build after pulling [uasm](https://terraspace.co.uk/uasm.html)?

Reply:
What does pulling usam entail?

Reply:
Also I didn't realize this was for `mingw-no-asm`

Reply:
Pulling uasm would entail getting a static `uasm64.exe`, which could be renamed to just `uasm.exe`  and using that as the drop-in replacement for compiling MASM.

More specifically, I was thinking of having github create a folder `uasm`, download the zip file, unpack `uasm64.exe` as `uasm.exe`, cache the folder, and then add that folder to PATH.

Reply:
What about https://github.com/egor-tensin/setup-mingw does this action properly setup the environment you expect?

Reply:
NOTE: Ideally you would change `mingw` to run on a Linux host to better match your situation.

Reply:
As a clarification, I normally compile this on Windows, but I usually code and fix from Linux, and decided to try out cross-compiling as I did not want to shuffle code back and forth from my linux machine to my windows machine.

> What about https://github.com/egor-tensin/setup-mingw does this action properly setup the environment you expect?

As far as I could tell, that action only sets up a mingw-w64 compiler, a `ml.exe` replacement, so that wouldn't work.

My normal situation is using msys2 to provide uasm and toolchain. One option might be to use the msys2 install that's available at `C:\msys64` to only pull `uasm` and add that to the PATH?

Reply:
> That sounds like a good approach.

I will start working on this. It may take some time as it's been a while since I last messed around with github actions

Reply:
Debian doesn't seem to provide uasm. Would you try yasm?

https://packages.debian.org/stable/allpackages?format=txt.gz